### PR TITLE
Feature/gql gs

### PR DESCRIPTION
**Pull request by gavinspr**
add hatching functionality and merge mui

### CODE REVIEW

File: src/components/MetaMaskConnection/MetaMaskConnection.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import ConnectButton from "./components/ConnectButton";
+import ConnectedMenu from "./components/ConnectedMenu";
+import { ethereum } from "../../common/types";
+
+async function checkCorrectNetwork(chainId: string) {
```
Comment (spope851) [inline src/components/MetaMaskConnection/MetaMaskConnection.tsx:6]: @gavinspr using this?

Comment (spope851) [inline src/components/MetaMaskConnection/MetaMaskConnection.tsx:6]: @gavinspr are you using ```chainId``` here? maybe to check if you're on the right chain already?

File: src/components/MetaMaskConnection/MetaMaskConnection.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      method: "eth_requestAccounts",
+    });
+
+    if (accounts.length > 0) {
+      await checkCorrectNetwork(ethereum.chainId);
+
+      const wallet = accounts[0];
+      setUserWallet(wallet);
+
+      return;
```
Comment (spope851) [inline src/components/MetaMaskConnection/MetaMaskConnection.tsx:29]: @gavinspr would it be useful for this one to return the wallet incase we want to get that somewhere else in the app and use it quickly not in state

File: src/components/all-legends/all-legends.tsx [hunk 3]
```diff
@@ -76,11 +76,11 @@ export const AllLegends: React.FC = () => {
 
     // console.log(legendSale);
 
-    // const legendOffer = await marketplaceContract.makeLegendOffer(
-    //   nftContract,
-    //   legendId,
-    //   { value: price }
-    // );
+    const legendOffer = await marketplaceContract.makeLegendOffer(
```
Comment (spope851) [inline src/components/all-legends/all-legends.tsx:79]: you can just run the function without instantiating it with const if you don't need ```legendOffer``` for anything. but maybe we should return legendOffer incase an error comes back?

File: src/components/AllLegends.tsx [hunk 4]
```diff
@@ -95,13 +95,14 @@ export const AllLegends = ({ itemsPerPage }: AllLegendsProps) => {
       </HStack>
       <SimpleGrid columns={3} spacing={10} pt={100}>
         {data &&
-          [...Array(legendTotal)].map((empty: any, legendId: any) => (
+          [...Array(21)].map((empty: any, legendId: any) => (
```
Comment (spope851) [inline src/components/AllLegends.tsx:98]: @gavinspr  try ```Record<string, never>``` for the type of ```empty``` and ```number``` for ```legendId```

**Review by spope851 — APPROVED** [review]
@gavinspr added a couple suggestions but all around this is solid. If you are all set with the code in the main branch I'll create a new main branch from what is currently in develop and deploy that to cloudflare before we merge this pr. lmk if that sounds good

File: src/components/MetaMaskConnection/MetaMaskConnection.tsx (hunk 1 above)
Comment (gavinspr) [reply src/components/MetaMaskConnection/MetaMaskConnection.tsx:6]: Ah yeah I didn't  need to pass that in, I removed it.

File: src/components/MetaMaskConnection/MetaMaskConnection.tsx (hunk 2 above)
Comment (gavinspr) [reply src/components/MetaMaskConnection/MetaMaskConnection.tsx:29]: I was thinking of putting it in context, and was about to convert the metamask component from chakra to mui. But I'm open to other solutions than context if you had some in mind.

File: src/components/all-legends/all-legends.tsx (hunk 3 above)
Comment (gavinspr) [reply src/components/all-legends/all-legends.tsx:79]: The write calls return an object with some data (hash, blocknumber, etc) which we may or may not want to use in the future. So either way, we can always change it back if we choose to use the returned object

**Pull request merged**